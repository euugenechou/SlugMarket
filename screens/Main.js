import React from 'react';
import { createBottomTabNavigator } from 'react-navigation';

import Explore from "./Explore";
import Profile from "./Profile";
import Settings from "./Settings";
import Icon from "react-native-vector-icons/Ionicons";

export default class Main extends React.Component {
  render () {
    return <BottomTabNavigator />;
  }
}

const BottomTabNavigator = createBottomTabNavigator({
  Explore: {
    screen: Explore,
    navigationOptions: {
      tabBarLabel: 'EXPLORE',
      tabBarIcon: ({ tintColor }) => (
        <Icon name="ios-search-outline"
          color={tintColor}
          size={24} />
      )
    }
  },
  Profile: {
    screen: Profile,
    navigationOptions: {
      tabBarLabel: 'PROFILE',
      tabBarIcon: ({ tintColor }) => (
        <Icon name="ios-person-outline"
          color={tintColor}
          size={24} />
      )
    }
  },
  Settings: {
    screen: Settings,
    navigationOptions: {
      tabBarLabel: 'SETTINGS',
      tabBarIcon: ({ tintColor }) => (
        <Icon name="ios-settings-outline"
          color={tintColor}
          size={24} />
      )
    }
  }
},
  {
    tabBarOptions: {
      activeTintColor: 'teal',
      inactiveTintColor: 'grey',
      style: {
        backgroundColor: 'white',
        borderTopWidth: 0,
        shadowOffset: { width: 5, height: 3 },
        shadowColor: 'black',
        shadowOpacity: 0.5,
        elevation: 5
      }
    }
});
