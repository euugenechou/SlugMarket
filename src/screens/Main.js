/* React imports */
import React from "react";
import {
  createBottomTabNavigator,
  createStackNavigator
} from "react-navigation";
import Icon from "react-native-vector-icons/Ionicons";

/* Local imports */
import Explore from "./Explore";
import AddItem from "./AddItem";
import ItemCategory from "./components/Explore/ItemCategory";
import ListingInfo from "./components/Explore/ListingInfo";

import Profile from "./Profile";
import EditProfile from "./EditProfile";
import UserListings from "./components/Profile/UserListings";
import UserListingInfo from "./components/Profile/UserListingInfo";

import Settings from "./Settings";
import SignIn from "./login/SignIn";
import Chat from "./Chat";

export default class Main extends React.Component {
  render() {
    return <BottomTabNavigator />;
  }
}

const ExploreStack = createStackNavigator(
  {
    MainExplore: {
      screen: Explore,
      headerVisible: false
    },
    AddItem: { screen: AddItem },
    ListingInfo: { screen: ListingInfo },
    ViewCategory: { screen: ItemCategory }
  },
  // { headerMode: 'none' }
);

const ProfileStack = createStackNavigator(
  {
    MainProfile: { 
      screen: Profile,
    },
    EditProfile: { screen: EditProfile },
    UserListingScreen: { screen: UserListings},
    UserListingInfoScreen: { screen: UserListingInfo}
  },
  // { headerMode: "none" }
);

const SettingsStack = createStackNavigator(
  {
    MainSettings: { screen: Settings },
    SignInScreen: { screen: SignIn }
  },
  // { headerMode: "none" }
);

const ChatStack = createStackNavigator(
  {
    MainChat: { screen: Chat},
  },
  {
      headerMode: "none",
  }
)

const BottomTabNavigator = createBottomTabNavigator(
  {
    Explore: {
      screen: ExploreStack,
      navigationOptions: {
        tabBarLabel: "EXPLORE",
        tabBarIcon: ({ tintColor }) => (
          <Icon name="ios-search-outline" color={tintColor} size={24} />
        )
      }
    },
    Profile: {
      screen: ProfileStack,
      navigationOptions: {
        tabBarLabel: "PROFILE",
        tabBarIcon: ({ tintColor }) => (
          <Icon name="ios-person-outline" color={tintColor} size={24} />
        )
      }
    },
    Chat: {
      screen: ChatStack,
      navigationOptions: {
        tabBarLabel: "MESSAGES",
        tabBarIcon: ({tintColor}) => (
          <Icon name = "ios-chatbubbles-outline" color ={tintColor} size = {24}/>
        )
      }
    },
    Settings: {
      screen: SettingsStack,
      navigationOptions: {
        tabBarLabel: "SETTINGS",
        tabBarIcon: ({ tintColor }) => (
          <Icon name="ios-settings-outline" color={tintColor} size={24} />
        )
      }
    }
  },
  {
    tabBarPosition: "bottom",
    swipeEnabled: "true",
    tabBarOptions: {
      activeTintColor: "teal",
      inactiveTintColor: "grey",
      style: {
        backgroundColor: "white",
        borderTopWidth: 0,
        shadowOffset: { width: 5, height: 3 },
        shadowColor: "black",
        shadowOpacity: 0.5,
        elevation: 5,
        paddingTop: 5,
        paddingBottom: 5
      }
    }
  }
);
