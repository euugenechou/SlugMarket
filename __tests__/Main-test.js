// __tests__/Main-test.js
import 'react-native';
import React from 'react';
import Main from '../src/screens/Main';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Main />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});