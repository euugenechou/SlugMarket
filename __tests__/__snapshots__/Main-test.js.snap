// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="auto"
      removeClippedSubviews={false}
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "opacity": 1,
            },
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column-reverse",
                },
                Object {
                  "backgroundColor": "#000",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "backgroundColor": "#E9E9EF",
                    "bottom": 0,
                    "left": 0,
                    "marginTop": -64,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "shadowColor": "black",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 5,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                    ],
                  }
                }
              >
                <RCTSafeAreaView
                  style={
                    Object {
                      "backgroundColor": "white",
                      "flex": 1,
                    }
                  }
                >
                  <RCTScrollView
                    refreshControl={
                      <RefreshControlMock
                        onRefresh={[Function]}
                        refreshing={false}
                      />
                    }
                    scrollEventThrottle={16}
                    showsVerticalScrollIndicator={false}
                    style={
                      Object {
                        "marginTop": 20,
                      }
                    }
                  >
                    <RCTRefreshControl />
                    <View>
                      <View
                        style={
                          Object {
                            "backgroundColor": "white",
                            "flex": 1,
                            "paddingTop": 40,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "fontSize": 32,
                              "fontWeight": "700",
                              "paddingBottom": 30,
                              "paddingHorizontal": 20,
                            }
                          }
                        >
                          Hi 
                          
                          ! 
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "fontSize": 24,
                              "fontWeight": "700",
                              "paddingHorizontal": 20,
                            }
                          }
                        >
                          Browse Items By Category
                        </Text>
                        <View
                          style={
                            Object {
                              "height": 130,
                              "marginTop": 20,
                            }
                          }
                        >
                          <RCTScrollView
                            horizontal={true}
                            showsHorizontalScrollIndicator={false}
                          >
                            <View>
                              <View
                                accessible={true}
                                isTVSelectable={true}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "white",
                                      "borderColor": "#dddddd",
                                      "borderRadius": 3,
                                      "borderWidth": 1,
                                      "elevation": 1,
                                      "height": 120,
                                      "marginLeft": 20,
                                      "shadowColor": "black",
                                      "shadowOffset": Object {
                                        "height": 0,
                                        "width": 0,
                                      },
                                      "shadowOpacity": 0.2,
                                      "width": 130,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "flex": 2,
                                      }
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "testUri": "../../../src/assets/textbooks.jpg",
                                        }
                                      }
                                      style={
                                        Object {
                                          "borderRadius": 3,
                                          "borderWidth": 0.1,
                                          "flex": 1,
                                          "height": null,
                                          "resizeMode": "cover",
                                          "shadowColor": "black",
                                          "shadowOffset": Object {
                                            "height": 0,
                                            "width": 0,
                                          },
                                          "shadowOpacity": 0.2,
                                          "width": null,
                                        }
                                      }
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "flex": 1,
                                        "justifyContent": "center",
                                      }
                                    }
                                  >
                                    <Text
                                      accessible={true}
                                      allowFontScaling={true}
                                      ellipsizeMode="tail"
                                      style={
                                        Object {
                                          "fontWeight": "600",
                                        }
                                      }
                                    >
                                       
                                      Textbooks
                                       
                                    </Text>
                                  </View>
                                </View>
                              </View>
                              <View
                                accessible={true}
                                isTVSelectable={true}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "white",
                                      "borderColor": "#dddddd",
                                      "borderRadius": 3,
                                      "borderWidth": 1,
                                      "elevation": 1,
                                      "height": 120,
                                      "marginLeft": 20,
                                      "shadowColor": "black",
                                      "shadowOffset": Object {
                                        "height": 0,
                                        "width": 0,
                                      },
                                      "shadowOpacity": 0.2,
                                      "width": 130,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "flex": 2,
                                      }
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "testUri": "../../../src/assets/furniture.jpg",
                                        }
                                      }
                                      style={
                                        Object {
                                          "borderRadius": 3,
                                          "borderWidth": 0.1,
                                          "flex": 1,
                                          "height": null,
                                          "resizeMode": "cover",
                                          "shadowColor": "black",
                                          "shadowOffset": Object {
                                            "height": 0,
                                            "width": 0,
                                          },
                                          "shadowOpacity": 0.2,
                                          "width": null,
                                        }
                                      }
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "flex": 1,
                                        "justifyContent": "center",
                                      }
                                    }
                                  >
                                    <Text
                                      accessible={true}
                                      allowFontScaling={true}
                                      ellipsizeMode="tail"
                                      style={
                                        Object {
                                          "fontWeight": "600",
                                        }
                                      }
                                    >
                                       
                                      Furniture
                                       
                                    </Text>
                                  </View>
                                </View>
                              </View>
                              <View
                                accessible={true}
                                isTVSelectable={true}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={null}
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "white",
                                      "borderColor": "#dddddd",
                                      "borderRadius": 3,
                                      "borderWidth": 1,
                                      "elevation": 1,
                                      "height": 120,
                                      "marginLeft": 20,
                                      "shadowColor": "black",
                                      "shadowOffset": Object {
                                        "height": 0,
                                        "width": 0,
                                      },
                                      "shadowOpacity": 0.2,
                                      "width": 130,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "flex": 2,
                                      }
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "testUri": "../../../src/assets/electronics.jpg",
                                        }
                                      }
                                      style={
                                        Object {
                                          "borderRadius": 3,
                                          "borderWidth": 0.1,
                                          "flex": 1,
                                          "height": null,
                                          "resizeMode": "cover",
                                          "shadowColor": "black",
                                          "shadowOffset": Object {
                                            "height": 0,
                                            "width": 0,
                                          },
                                          "shadowOpacity": 0.2,
                                          "width": null,
                                        }
                                      }
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "flex": 1,
                                        "justifyContent": "center",
                                      }
                                    }
                                  >
                                    <Text
                                      accessible={true}
                                      allowFontScaling={true}
                                      ellipsizeMode="tail"
                                      style={
                                        Object {
                                          "fontWeight": "600",
                                        }
                                      }
                                    >
                                       
                                      Electronics
                                       
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </RCTScrollView>
                        </View>
                        <View
                          style={
                            Object {
                              "marginTop": 20,
                            }
                          }
                        >
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "fontSize": 24,
                                "fontWeight": "700",
                                "paddingHorizontal": 20,
                              }
                            }
                          >
                            Recently Added Items
                          </Text>
                          <View
                            style={
                              Object {
                                "flexDirection": "column",
                                "flexWrap": "wrap",
                                "justifyContent": "space-evenly",
                                "marginBottom": 20,
                                "paddingHorizontal": 20,
                                "paddingVertical": 20,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </RCTScrollView>
                </RCTSafeAreaView>
              </View>
            </View>
            <View
              onLayout={[Function]}
              pointerEvents="box-none"
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F7F7F7",
                    "transform": Array [
                      Object {
                        "translateX": -750,
                      },
                    ],
                  }
                }
              >
                <View
                  onLayout={[Function]}
                  pointerEvents="box-none"
                  style={
                    Object {
                      "backgroundColor": "#F7F7F7",
                      "borderBottomColor": "#A7A7AA",
                      "borderBottomWidth": 0.5,
                      "height": 64,
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 20,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  />
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Object {
        "backgroundColor": "white",
        "borderTopColor": "#dddddd",
        "borderTopWidth": 0.5,
        "elevation": 5,
        "flexDirection": "row",
        "height": 49,
        "paddingBottom": 5,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 5,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 3,
          "width": 5,
        },
        "shadowOpacity": 0.5,
      }
    }
  >
    <View
      accessibilityLabel="EXPLORE"
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
          },
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "flexDirection": "column",
            "justifyContent": "flex-end",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 29,
            },
            false,
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "height": "100%",
              "justifyContent": "center",
              "minWidth": 25,
              "opacity": 1,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "teal",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "height": "100%",
              "justifyContent": "center",
              "minWidth": 25,
              "opacity": 0,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "grey",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "teal",
            "fontSize": 10,
            "marginBottom": 1.5,
            "textAlign": "center",
          }
        }
      >
        EXPLORE
      </Text>
    </View>
    <View
      accessibilityLabel="PROFILE"
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
          },
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "flexDirection": "column",
            "justifyContent": "flex-end",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 29,
            },
            false,
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "height": "100%",
              "justifyContent": "center",
              "minWidth": 25,
              "opacity": 0,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "teal",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "height": "100%",
              "justifyContent": "center",
              "minWidth": 25,
              "opacity": 1,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "grey",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "grey",
            "fontSize": 10,
            "marginBottom": 1.5,
            "textAlign": "center",
          }
        }
      >
        PROFILE
      </Text>
    </View>
    <View
      accessibilityLabel="SETTINGS"
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
          },
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "flexDirection": "column",
            "justifyContent": "flex-end",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 29,
            },
            false,
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "height": "100%",
              "justifyContent": "center",
              "minWidth": 25,
              "opacity": 0,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "teal",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "height": "100%",
              "justifyContent": "center",
              "minWidth": 25,
              "opacity": 1,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "grey",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "grey",
            "fontSize": 10,
            "marginBottom": 1.5,
            "textAlign": "center",
          }
        }
      >
        SETTINGS
      </Text>
    </View>
  </View>
</View>
`;
